src/Core/Foundation/Version.php

```diff
@@ -242,12 +242,14 @@ private function versionCompare($version, $operator)
      */
     private function checkVersion($version)
     {
-        if (!is_string($version)) {
-            throw InvalidVersionException::mustBeAString();
-        }
-
-        if (!preg_match('/^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$/', $version)) {
-            throw InvalidVersionException::mustBeValidName();
+        if (!preg_match('~^\d+(\.\d+){0,}$~', $version)) {
+            throw new InvalidVersionException(
```

Comment:
It's better to get exception messages - when not translated - into exception classes.

Reply:
Agree but use `protected $message = ''; ` instead static call :)